src/controllers/sign.out.controller.ts

```diff
@@ -29,17 +28,34 @@ export const post = (req: Request, res: Response, next: NextFunction) => {
     logger.debugRequest(req, `POST ${config.SIGN_OUT_PAGE}`);
     const previousPage = req.body["previousPage"];
 
-    if (!previousPage.includes(config.REGISTER_AN_OVERSEAS_ENTITY_URL)){
+    if (!previousPage.startsWith(config.REGISTER_AN_OVERSEAS_ENTITY_URL)){
       throw createAndLogErrorRequest(req, `${previousPage} page is not part of the journey!`);
     }
 
     if (req.body["sign_out"] === 'yes') {
       return res.redirect(config.ACCOUNTS_SIGN_OUT_URL);
     }
 
-    return res.redirect(previousPage);
+    return safeRedirect(res, previousPage);
   } catch (error) {
     logger.errorRequest(req, error);
     next(error);
   }
 };
+
+function getPreviousPageUrl(req: Request) {
+  const headers = req.rawHeaders;
+  const absolutePreviousPageUrl = headers.filter(item => item.includes(config.REGISTER_AN_OVERSEAS_ENTITY_URL))[0];
+
+  // Don't attempt to determine a relative previous page URL if no absolute URL is found
+  if (!absolutePreviousPageUrl) {
+    return absolutePreviousPageUrl;
+  }
+
+  const startingIndexOfRelativePath = absolutePreviousPageUrl.indexOf(config.REGISTER_AN_OVERSEAS_ENTITY_URL);
```

Comment:
what happens if the absolutePreviousPageUrl doesn't contain the REGISTER_AN_OVERSEAS_ENTITY_URL, i think it will return '-1', which will be passed into substring - will that cause an error?

Reply:
I have just tested this. It is fine because in the case that `absolutePreviousPageUrl` doesn't contain the REGISTER_AN_OVERSEAS_ENTITY_URL, the function will return at line 52. It will never get to the `indexOf` or `substring` calls.

I've added a new unit test to prove this though.